Homework/app/controllers/cleaners_controller.rb

```diff
@@ -20,21 +27,26 @@ def create
     @cities = []
     @cleaner = Cleaner.new(cleaner_params)
       if @cleaner.save
-
+        params[:city_ids].each do |id|
+          CitiesCleaner.create(city_id:id,cleaner_id:@cleaner.id)
+        end
         redirect_to @cleaner
       else
         render :new
       end
   end
 
   def edit
-    @cleaner = Cleaner.find(params[:id])
-    @cities = []
-    @cleaner.cities.each {|city| @cities.push(city.id)}
+    if @cleaner.present?
+      @cities = []
```

Comment:
@neelgohel instead you can use `.inject`, that would allow to avoid declaring this empty array variable.